hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -151,6 +155,14 @@ protected AbstractHoodieLogRecordReader(FileSystem fs, String basePath, List<Str
     this.instantRange = instantRange;
     this.withOperationField = withOperationField;
     this.enableFullScan = enableFullScan;
+
+    // virtual keys handling
+    if (!tableConfig.populateMetaFields()) {
+      this.virtualKeysEnabled = true;
```

Comment:
Seems like virtualKeysEnabled will always be equal to tableConfig.populateMetaFields(). Can we remove this additional field?

Reply:
virtualKeysEnabled field is removed. But, we either need to retain the table config or populateMetaField value from the config so that few methods like scan(), createHoodieRecord know can act accordingly. For now going with the variable of same name.